src/st.py

```diff
@@ -44,22 +33,23 @@ def SuffixTree(string):
     tree = SuffixTree(ref)
     read = 'aba'
     print([t for t in bf_order(tree)])
-    
+    #>>> [[None, None, None], [0, 1, None], [1, 3, None], [6, 7, 6], [3, 7, 2], [1, 3, None], [6, 7, 5], [6, 7, 4], [3, 7, 1], [6, 7, 3], [3, 7, 0]]
     '''
     if string == '' or string == None:
-        return []
+        return None
 
     string += '$'  # add sentinal to string.
-    tree = Node(None,None,None)  # create root.
-    count = 0  # enables tracking of longest path (since we iterate left->right each iteration will continuously add longest->shortest path).
+    tree = Node(None,None)  # create root.
+    
+    count = 0  # enables tracking of longest suffix (since we iterate left->right each iteration will continuously add longest->shortest suffix).
     for i in range(len(string)):  # loop through all suffixes.
         #print(count)
         current = tree  # set root as starting point.
         j = i # set i as startpoint for j and use k (see below) to increment j inside (nodes) while loop.
         while j < len(string):  # from the root walk down as far as possible.
             if string[j] in current.out:  # if a child contains 'first' letter, go that direction.
                 child = current.out[string[j]]  
-                val = repString(string, child.start, child.end)
+                val = string[child.start:child.end]
```

Comment:
You are slicing the string. If that is on purpose, then that is fine. If not, now you know.